### PR TITLE
Prefer `seq_mem`s to `comb_mem`s all over

### LINKED ISSUE #2062: Combinational memories in `gen_exp` and `gen_ntt_pipeline`
In exactly two places, we use the eDSL to generate combinational and not sequential memories. These are `gen_exp` and `gen_ntt_pipeline`. We have discussed that we should use default to sequential memories, and use combinational memories only when we know that the memory is very small. I just wanted to flag these two cases in case we want to revisit them. Porting them over will be very easy.

_Originally posted by @anshumanmohan in https://github.com/calyxir/calyx/issues/2052#issuecomment-2128007259_

### PR DESCRIPTION
**Pull request by anshumanmohan**
This PR edits `gen_exp.py` and `gen_ntt_pipeline.py`, the last remaining files that used the eDSL to create combinational memories, so that they use sequential memories instead. 

Closes #2062. As expected, the fix was easy.

The changes to the .py files are straightforward. There is a largish diff because we are expect-testing the Calyx files generated (where you'll see changed lines of Calyx) and the results of simulation (where you'll see slightly elevated cycle counts).

### CODE REVIEW

**Comment by anshumanmohan** [comment]
Requesting a review from @KabirSamsi since you may be in the business of `comb`-to-`seq` conversion yourself!

File: frontends/ntt-pipeline/gen-ntt-pipeline.py [hunk 1]
```diff
@@ -207,10 +207,12 @@ def preamble_group(comp: cb.ComponentBuilder, row):
...
-            reg.in_ = input.read_data
-            phi.write_en = 1
-            phi.in_ = phis.read_data
+            input.content_en = 1
+            phis.content_en = 1
+            reg.write_en = input.done @ 1
+            reg.in_ = input.done @ input.read_data
+            phi.write_en = input.done @ 1
+            phi.in_ = phis.done @ phis.read_data
             preamble.done = (reg.done & phi.done) @ 1
```
Comment (anshumanmohan) [inline frontends/ntt-pipeline/gen-ntt-pipeline.py:216]: @KabirSamsi this is a case of [guard conjunction](https://docs.calyxir.org/lang/ref.html#guards) in the wild. @rachitnigam I'm guessing looks pretty risky to you haha

Comment (rachitnigam) [reply frontends/ntt-pipeline/gen-ntt-pipeline.py:216]: Yup! Pretty risky. I would restructure this as two static groups that write to the memory and run in parallel instead.

Comment (anshumanmohan) [reply frontends/ntt-pipeline/gen-ntt-pipeline.py:216]: Alrighty, there is now https://github.com/calyxir/calyx/pull/2131

Comment (KabirSamsi) [reply frontends/ntt-pipeline/gen-ntt-pipeline.py:216]: Should we keep reviewing this PR, or review #2131 instead?

Comment (anshumanmohan) [reply frontends/ntt-pipeline/gen-ntt-pipeline.py:216]: You can focus on this one Kabir, but we'll keep both PRs! There will be a minor merge conflict when one of these two is merged into main and the other tries to merge into the new main, but I'll deal with that.

Comment (anshumanmohan) [reply frontends/ntt-pipeline/gen-ntt-pipeline.py:216]: In truth maybe I should have branched off this branch, and chosen this branch as the base of the merge, for 2131. Ah well. My bad, and I'll eat the merge conflict issues

Comment (anshumanmohan) [reply frontends/ntt-pipeline/gen-ntt-pipeline.py]: The other branch has been merged into main and this one has been caught up to those changes. Ready for a final look. Once you have completed the review process, Kabir, I'll hit merge.

**Comment by KabirSamsi** [comment]
Looks all good on my end! All tests seem to pass fine as well.

**Review by KabirSamsi — APPROVED** [review]
Looks all good on my end! All tests seem to pass fine as well.

**Pull request merged**